### PR TITLE
Capacity runtime api with list_unclaimed_rewards endpoint

### LINKED ISSUE #1698: Custom RPC to get info about rewards that are about to expire
## Story
As a user, I would like to know if I have unclaimed staking rewards and when they will expire.

### Acceptance criteria
- [x] implement `list_unclaimed_rewards` fn in the Capacity pallet.
- [ ] Unit and e2e Tests for same

### Dependencies
- [x] Ability to calculate a reward for an individual staking account for a given Reward Era

### PR DESCRIPTION
**Pull request by shannonwells**
# Goal
The goal of this PR is to implement a capacity runtime api for the list_unclaimed_rewards endpoint.

Following the current recommendation of Parity, there is no custom RPC matching this runtime API endpoint; instead, use state.call to access the endpoint.  For examples using either Observable or APIPromise, see the e2e tests.

Closes #1698 

# Checklist
- [x] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [x] Tests added

### CODE REVIEW

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -1518,6 +1519,15 @@ impl_runtime_apis! {
...
 
+	impl pallet_capacity_runtime_api::CapacityRuntimeApi<Block, AccountId, Balance, BlockNumber> for Runtime {
+		fn list_unclaimed_rewards(who: AccountId) -> Vec<UnclaimedRewardInfo<Balance, BlockNumber>> {
+			match Capacity::list_unclaimed_rewards(&who) {
+				Ok(rewards) => return rewards.into_inner(),
+				Err(_) => return Vec::new(),
+			}
+		}
+	}
+
```
Comment (saraswatpuneet) [inline runtime/frequency/src/lib.rs:1530]: nit: need to update `spec_version`?

**Review by saraswatpuneet — APPROVED** [review]
nice work!! Nothing unusual stood out, looks good to me

File: runtime/frequency/src/lib.rs (hunk 1 above)
Comment (shannonwells) [reply runtime/frequency/src/lib.rs:1530]: That will come with the merge to main.

Comment (saraswatpuneet) [reply runtime/frequency/src/lib.rs:1530]: I see it's going in feature branch 🫡

**Review by enddynayn — APPROVED** [review]
LGTM 👍

File: js/api-augment/definitions/capacity.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        BlockNumber: 'u32',
+        UnclaimedRewardInfo: {
+            reward_era: 'RewardEra',
+            expires_at_block: 'BlockNumber',
+            staked_amount: 'Balance',
+            eligible_amount: 'Balance',
+            earned_amount: 'Balance',
+        }
+    },
+    runtime: {
```
Comment (enddynayn) [inline js/api-augment/definitions/capacity.ts:17]: Nice. By the way, do this show up in PolkadotUI now?

File: pallets/capacity/src/runtime-api/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+[package]
+name = "pallet-capacity-runtime-api"
+version = "0.0.0"
+description = "A package that adds Runtime Api for the Stateful Storage pallet"
```
Comment (mattheworris) [inline pallets/capacity/src/runtime-api/Cargo.toml:4]:
```suggestion
description = "A package that adds Runtime Api for the Capaciy pallet"
```

**Review by mattheworris — APPROVED** [review]
- Read through changes
- Executed unit tests (All Passed)
- Executed e2e-test-serial (All passed)
- nit: one typo 
🚀 LGTM 🚀

File: js/api-augment/definitions/capacity.ts (hunk 2 above)
Comment (shannonwells) [reply js/api-augment/definitions/capacity.ts:17]: It won't until it merges to main and I think you still have to post a PR with the updated release to PolkadotJS ui

**Pull request merged**